engine/src/main/java/org/terasology/rendering/cameras/OpenVRStereoCamera.java

```diff
@@ -206,49 +211,68 @@ public void updateMatrices() {
         updateMatrices(activeFov);
     }
 
+    private void jomlMatrix4f(org.joml.Matrix4f matrixInput, org.terasology.math.geom.Matrix4f matrixOut) {
+        matrixOut.set(0, 0, matrixInput.m00());
+        matrixOut.set(0, 1, matrixInput.m10());
+        matrixOut.set(0, 2, matrixInput.m20());
+        matrixOut.set(0, 3, matrixInput.m30());
+        matrixOut.set(1, 0, matrixInput.m01());
+        matrixOut.set(1, 1, matrixInput.m11());
+        matrixOut.set(1, 2, matrixInput.m21());
+        matrixOut.set(1, 3, matrixInput.m31());
+        matrixOut.set(2, 0, matrixInput.m02());
+        matrixOut.set(2, 1, matrixInput.m12());
+        matrixOut.set(2, 2, matrixInput.m22());
+        matrixOut.set(2, 3, matrixInput.m32());
+        matrixOut.set(3, 0, matrixInput.m03());
+        matrixOut.set(3, 1, matrixInput.m13());
+        matrixOut.set(3, 2, matrixInput.m23());
+        matrixOut.set(3, 3, matrixInput.m33());
+    }
+
     @Override
     public void updateMatrices(float fov) {
         prevViewProjectionMatrix.set(viewProjectionMatrix);
 
-        // Nothing to do...
-        if (cachedPosition.equals(getPosition()) && cachedViewigDirection.equals(getViewingDirection())
-                && cachedZFar == zFar && cachedZNear == zNear) {
+        org.joml.Matrix4f leftEyeProjection = vrProvider.vrState.getEyeProjectionMatrix(0);
+        org.joml.Matrix4f rightEyeProjection = vrProvider.vrState.getEyeProjectionMatrix(1);
+        org.joml.Matrix4f leftEyePose = vrProvider.vrState.getEyePose(0);
+        org.joml.Matrix4f rightEyePose = vrProvider.vrState.getEyePose(1);
```

Comment:
I'm not sure I understood your comment here @indianajohn. Are you saying you can move the VRProvider in a module and consequentially remove JOML from Terasology? How much of the code you have written would relocate into the module alongside the VRProvider?

In any case I feel this can be left to a separate PR and we can leave things as they are for the time being, so that this PR just doesn't stay open for too long. But if my understanding is correct I'd really like for VR code to exist in a module rather than directly in the engine.